src/borg/helpers.py

```diff
@@ -1546,3 +1547,27 @@ def signal_handler(signo, handler):
         yield
     finally:
         signal.signal(signo, old_signal_handler)
+
+
+class ErrorFreeTextIOWrapper(io.TextIOWrapper):
+    def read(self, n):
+        if not self.closed:
+            try:
+                return super(ErrorFreeTextIOWrapper, self).read(n)
+            except IOError:
+                try:
+                    super(ErrorFreeTextIOWrapper, self).close()
+                except IOError:
+                    pass
+        return ''
+
+    def write(self, s):
+        if not self.closed:
+            try:
+                return super(ErrorFreeTextIOWrapper, self).write(s)
+            except IOError:
+                try:
+                    super(ErrorFreeTextIOWrapper, self).close()
+                except IOError:
```

Comment:
One minor nitpick, quoting from the python 3.4 docs:
```
The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of OSError.

exception EnvironmentError

exception IOError

exception WindowsError
```

So, please use OSError at all these places.